kwave/utils/signals.py

```diff
@@ -142,8 +142,12 @@ def cosine_series(n: int, N: int, coeffs: List[float]) -> np.ndarray:
 
     # create the window
     if N.size == 1:
+        # TODO: what should this behaviour be if N is a list of ints? make windows of multiple lengths?
```

Comment:
Why are you concerned for N being list of ints? Based on the function definition, I think N is expected to be either an int or an array with a single int element if `dim=1`.